src/components/ConnectButtonInner.tsx

```diff
@@ -0,0 +1,33 @@
+import { Button } from '@chakra-ui/react';
+import { Trans } from '@lingui/macro';
+import type { FunctionComponent } from 'react';
+
+import { ConnectButtonAvatar } from './ConnectButtonAvatar';
+
+export type ConnectButtonInnerProps = {
+  handleOnClick?: (() => void) | undefined;
+  isConnected: boolean;
+  ensName?: string | undefined;
+  truncatedAddress?: string | undefined;
+};
+
+export const ConnectButtonInner: FunctionComponent<ConnectButtonInnerProps> = ({
+  handleOnClick,
+  isConnected,
+  ensName,
+  truncatedAddress,
+}) => {
+  return (
+    <Button
+      variant={isConnected ? 'connected' : 'connect'}
+      leftIcon={
+        <ConnectButtonAvatar isConnected={isConnected} isFallback={!ensName} />
+      }
+      iconSpacing={isConnected ? '0' : '2'} // TODO: Add Space for Avatar Icon if Avatar Icon is ready
+      onClick={handleOnClick}
+      width={isConnected ? '55%' : '50%'}
```

Comment:
I still don't understand why this `width` is necessary?

Reply:
i see your point, if we dont fix it, it will become 100% width

Reply:
resolve by using padding